product_code/user_service/impl/src/main/scala/de/upb/cs/uc4/user/impl/UserServiceImpl.scala

```diff
@@ -160,74 +171,84 @@ class UserServiceImpl(
   override def updateUser(username: String): ServerServiceCall[User, Done] =
     identifiedAuthenticated(AuthenticationRole.All: _*) {
       (authUsername, role) =>
-        ServerServiceCall { (_, userRaw) =>
-          val user = userRaw.clean
-          // Check, if the username in path is different than the username in the object
-          if (username != user.username) {
-            throw UC4Exception.PathParameterMismatch
-          }
-
-          // If invoked by a non-Admin, check if the manipulated object is owned by the user
-          if (role != AuthenticationRole.Admin && authUsername != user.username) {
-            throw UC4Exception.OwnerMismatch
-          }
-
-          //validate new user and check, if username errors exist, since entityRef might fail if username is incorrect
-          var validationErrors = user.validate
-          if (validationErrors.map(_.name).contains("username")) {
-            throw new UC4Exception(422, DetailedError(ErrorType.Validation, validationErrors))
-          }
+        ServerServiceCall {
+          (_, userRaw) =>
+            val user = userRaw.clean
+            // Check, if the username in path is different than the username in the object
+            if (username != user.username) {
+              throw UC4Exception.PathParameterMismatch
+            }
 
-          val ref = entityRef(user.username)
-          ref.ask[Option[User]](replyTo => GetUser(replyTo)).flatMap { optUser =>
-            if (optUser.isEmpty) {
-              // Add to validation errors, and throw prematurely since uneditable fields are uncheckable
-              validationErrors :+= SimpleError("username", "Username not in use.")
-              throw new UC4Exception(422, DetailedError(ErrorType.Validation, validationErrors))
+            // If invoked by a non-Admin, check if the manipulated object is owned by the user
+            if (role != AuthenticationRole.Admin && authUsername != user.username) {
+              throw UC4Exception.OwnerMismatch
             }
-            val oldUser = optUser.get
-            // Other users than admins can only edit specified fields
-            var editErrors = oldUser.checkUneditableFields(user)
-            if (role != AuthenticationRole.Admin) {
-              editErrors ++= oldUser.checkProtectedFields(user)
+
+            //validate new user and check, if username errors exist, since entityRef might fail if username is incorrect
+            var validationErrors = try {
+              Await.result(user.validate, 5.seconds)
             }
-            if (editErrors.nonEmpty) {
-              throw new UC4Exception(422, DetailedError(ErrorType.UneditableFields, editErrors))
+            catch {
+              case _: TimeoutException => throw UC4Exception.ValidationTimeout
+              case e: Exception        => throw e
```

Comment:
This needs to be changed into a Internal Server Error
```suggestion
              case e: Exception        => throw UC4Exception.InternalServerError("Validation Error", e.getMessage)
```